apps/test/unit/MusicControllerTest.js

```diff
@@ -0,0 +1,72 @@
+import {expect} from '../util/reconfiguredChai';
+import MusicController from '@cdo/apps/MusicController';
+import sinon from 'sinon';
+import Sounds from '@cdo/apps/Sounds';
+import winMp3 from '!!file-loader!../audio/assets/win.mp3';
+
+describe('MusicController', () => {
+  const track = {
+    volume: 1,
+    hasOgg: false,
+    name: 'win',
+    isloaded: false,
+    group: 'mygroup'
+  };
+
+  let musicController, sound, sounds, sourceURL;
+
+  beforeEach(() => {
+    sounds = new Sounds();
+    sourceURL = winMp3;
+    sounds.register({
+      id: winMp3,
+      mp3: winMp3
+    });
+    sound = sounds.soundsById[sourceURL];
+    sinon.spy(sound, 'play');
+    sinon.stub(Sounds.prototype, 'registerByFilenamesAndID').returns(sound);
+  });
+
+  afterEach(() => {
+    sinon.restore();
```

Comment:
From [docs](https://sinonjs.org/releases/latest/stubs/): The original function can be restored by calling object.method.restore(); (or stub.restore();

So I think what you would want here is `Sounds.prototype.registerByFilenamesAndID.restore()` and `sound.play.restore()`

Reply:
🙌🏼